blacksheep/server/gzip.py

```diff
@@ -0,0 +1,154 @@
+import gzip
+import asyncio
+from concurrent.futures import Executor, ThreadPoolExecutor
+
+from blacksheep.server.normalization import ensure_response
+from blacksheep import Request, Response, Content
+from blacksheep.server.application import Application
+
+from typing import Callable, Awaitable, Optional, Iterable, List
+
+
+class GzipMiddleware:
+    """
+    The gzip compression middleware for all requests with a body larger than
+    the specified minimum size and with the "gzip" encoding in the "Accept-Encoding"
+    header.
+
+    Parameters
+    ----------
+    min_size: int
+        The minimum size of the response body to compress.
+    comp_level: int
+        The compression level to use.
+    handled_types: Optional[Iterable[bytes]]
+        The list of content types to compress.
+    executor: Executor
+        The executor to use for compression.
+    """
+
+    handled_types: List[bytes] = [
+        b"json",
+        b"xml",
+        b"yaml",
+        b"html",
+        b"text/plain",
+        b"application/javascript",
+        b"text/css",
+        b"text/csv",
+    ]
+
+    def __init__(
+        self,
+        min_size: int = 500,
+        comp_level: int = 5,
+        handled_types: Optional[Iterable[bytes]] = None,
+        executor: Executor = ThreadPoolExecutor,
```

Comment:
I guess we can spawn a thread pool and then shoot it down in `__del__` method in case users don't provide an instance themself. 

I'll implement such behavior and add a notice in the docstring. It looks smoother than making users initiate a new poll each time they want to use gzip compression. I don't think it should affect performance. 

@RobertoPrevato What do you think about the approach?

Reply:
I think we can use a default of `None` because `loop.run_in_executor` itself uses a default executor pool when none is provided. I recommend to follow the same approach of that built-in function.

Reply:
@RobertoPrevato yep, it sounds reasonable